runner-readme.md

```diff
@@ -0,0 +1,216 @@
+# Kola Runner
+
+## Table of Contents
+
+ * [General Function Overview](#general-function-overview)
+ * [Visual Function Flow](#visual-function-flow)
+ * [Detailed Workflow](#detailed-workflow)
+ * [Common Closures](#common-closures)
+ * [Logging](#logging)
+
+The `kola` runner is very similar to the standard `go test` runner with some
+modifications to allow tighter control over the run loop and extensibility
+for logging (without parsing) & output control.
+
+## General Function Overview
+
+### kola/harness
+
+#### kola/harness: RunTests
+
+Responsible for filtering down the test list based on the given criteria,
+creating the `harness/suite: Suite` object, and outputting the final result.
+
+#### kola/harness: runTest
+
+Creates the test cluster, runs the individual test function, and cleans up the
+test cluster.
+
+### harness/suite
+
+#### harness/suite: Run
+
+Creates the output directory, the `test.tap` file and any profile related
+files then calls `harness/suite: runTests`.
+
+#### harness/suite: runTests
+
+Sets up the `harness/harness: H` object and calls `harness/harness: tRunner`
+with a closure to call `harness/harness: Run` for each test.
+
+### harness/harness
+
+#### harness/harness: tRunner
+
+Responsible for the timing, reporting, and execution of a closure.
+
+#### harness/harness: Run
+
+Handles the setup of child `harness/harness: H` objects, loggers, and the
+running of closures as subtests.
+
+## Visual Function Flow
+
+```
+cmd/kola/kola
```

Comment:
would be helpful to have "contents" between the arrows, i.e. unfiltered tests going into runTests and a test suite with applicable tests coming out

Reply:
Would also be helpful to break this into chunks. E.g. the first N steps do X, next M do Y, etc